oomph/typer.py

```diff
@@ -256,15 +248,17 @@ def do_statement(self, ast: uast.Statement) -> List[tast.Statement]:
         if isinstance(ast, uast.Let):
             assert ast.varname not in self.variables, (ast.varname, self.variables)
             value = self.do_expression(ast.value)
-            self.variables[ast.varname] = value.type
-            return [tast.CreateLocalVar(ast.varname, value)]
+            var = tast.LocalVariable(ast.varname, value.type)
+            self.variables[var.name] = var
+            return [tast.CreateLocalVar(var, value)]
 
         if isinstance(ast, uast.Assign):
             # TODO: this assumes local variable without assert
-            vartype = self.variables[ast.varname]
+            var2 = self.variables[ast.varname]  # mypy_sucks you mypy
```

Comment:
This is a find/replace problem:

```suggestion
            var2 = self.variables[ast.varname]  # fuck you mypy
```